hack/tools/tilt-prepare/main.go

```diff
@@ -0,0 +1,486 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/pflag"
+	"golang.org/x/net/context"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/kubernetes/scheme"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	clusterctlv1 "sigs.k8s.io/cluster-api/cmd/clusterctl/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/config"
+	"sigs.k8s.io/cluster-api/test/infrastructure/container"
+	utilyaml "sigs.k8s.io/cluster-api/util/yaml"
+)
+
+/*
+Example call for tilt up:
+	--tools kustomize,envsubst
+	--cert-manager
+	--kustomize-builds clusterctl.crd:./cmd/clusterctl/config/crd/
+	--kustomize-builds observability.tools:./hack/observability/
+	--providers core:.:debug
+	--providers kubeadm-bootstrap:./bootstrap/kubeadm
+	--providers kubeadm-control-plane:./controlplane/kubeadm
+	--providers docker:./test/infrastructure/docker
+*/
+
+var (
+	rootPath      string
+	tiltBuildPath string
+
+	toolsFlag           = pflag.StringSlice("tools", []string{}, "list of tools to be created; each value should correspond to a make target")
+	certManagerFlag     = pflag.Bool("cert-manager", false, "prepare cert-manager")
+	kustomizeBuildsFlag = pflag.StringSlice("kustomize-builds", []string{}, "list of kustomize build to be run; each value should be in the form name:path")
+	providersBuildsFlag = pflag.StringSlice("providers", []string{}, "list of providers to be installed; each value should be in the form name:path[:debug]")
+)
+
+const (
+	kustomizePath = "./hack/tools/bin/kustomize"
+	envsubstPath  = "./hack/tools/bin/envsubst"
+)
+
+func init() {
+	// TODO: init root dynamically
+	rootPath = "/Users/fpandini/go/src/sigs.k8s.io/cluster-api"
+	tiltBuildPath = filepath.Join(rootPath, ".tiltbuild")
+
+	_ = clusterctlv1.AddToScheme(scheme.Scheme)
+}
+
+// This tool aims to speed up tilt startup time by running in parallel a set of task
+// preparing everything is required for tilt up.
+func main() {
+	fmt.Printf("[main] started\n")
+	start := time.Now()
+
+	pflag.Parse()
+
+	// Execute a first group of tilt prepare tasks, building all the tools required in subsequent steps/by tilt.
+	if err := tiltTools(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	// execute a second group of tilt prepare tasks, building all the resources required by tilt.
+	if err := tiltResources(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
```

Comment:
klog.Fatal()

Reply:
klog.Fatal creates a panic, which gives a bad ux in tilt dashboard